app/routes.php

```diff
@@ -77,7 +77,7 @@ function( Request $request ) use ( $app, $ffFactory ) {
 $app->post(
 	'validate-address', // Validates donor information. This route is named badly.
 	function( Request $request ) use ( $app, $ffFactory ) {
-		return ( new ValidateAddressHandler( $ffFactory, $app ) )->handle( $request );
+		return ( new ValidateDonorHandler( $ffFactory, $app ) )->handle( $request );
```

Comment:
Just a side note: the `/validate-address` route is also called by the membership form. Since the URL and the underlying code did not change substantially, it will continue to work, but calling `ValidateDonorHandler` could make it weird, especially since the `ApplyForMembershipHandler` does its own address data validation, which should also not be different from `/validate-address`.

Reply:
Ugggggghhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh

How about having a `/validate-donor` and a `/validate-applicant`?

Reply:
Yes, two routes would be a clean solution. Although it's not as DRY as I like, I could live with 100% identical code in the two validators, because they are two contexts.